nifi-nar-bundles/nifi-prometheus-bundle/nifi-prometheus-reporting-task/src/main/java/org/apache/nifi/reporting/prometheus/PrometheusReportingTask.java

```diff
@@ -0,0 +1,105 @@
+@CapabilityDescription("")
+@DefaultSchedule(strategy = SchedulingStrategy.TIMER_DRIVEN, period = "1 sec")
+
+public class PrometheusReportingTask extends AbstractReportingTask {
+
+    private PrometheusServer prometheusServer;
+
+    static final PropertyDescriptor METRICS_ENDPOINT_PORT = new PropertyDescriptor.Builder().name("Prometheus Metrics Endpoint Port").description("The Port where prometheus metrics can be accessed")
+            .required(true).expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY).defaultValue("9092").addValidator(StandardValidators.INTEGER_VALIDATOR).build();
+
+    static final PropertyDescriptor APPLICATION_ID = new PropertyDescriptor.Builder().name("Application ID").description("The Application ID to be included in the metrics sent to Prometheus")
+            .required(true).expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY).defaultValue("nifi").addValidator(StandardValidators.NON_EMPTY_VALIDATOR).build();
+
+    static final PropertyDescriptor INSTANCE_ID = new PropertyDescriptor.Builder().name("Instance ID").description("Id of this NiFi instance to be included in the metrics sent to Prometheus")
+            .required(true).expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY).defaultValue("${hostname(true)}").addValidator(StandardValidators.NON_EMPTY_VALIDATOR).build();
+
+    static final PropertyDescriptor SEND_JVM_METRICS = new PropertyDescriptor.Builder().name("Send JVM-metrics").description("Send JVM-metrics in addition to the Nifi-metrics")
+            .allowableValues("true", "false").defaultValue("false").required(true).build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+
+        properties.add(METRICS_ENDPOINT_PORT);
+        properties.add(APPLICATION_ID);
+        properties.add(INSTANCE_ID);
+        properties.add(SEND_JVM_METRICS);
+
+        return properties;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ConfigurationContext context) {
+
+        final String metricsEndpointPort = context.getProperty(METRICS_ENDPOINT_PORT).getValue();
+
+        try {
+            this.prometheusServer = new PrometheusServer(new InetSocketAddress(Integer.parseInt(metricsEndpointPort)), getLogger());
+            getLogger().info("Started JETTY server");
+        } catch (Exception e) {
+            getLogger().error("Error: " + e);
+            e.printStackTrace();
+        }
+
+    }
+
+    @OnStopped
+    public void OnStopped() throws Exception {
+        Server server = prometheusServer.getServer();
+        server.stop();
+    }
+
+    @OnShutdown
+    public void onShutDown() throws Exception {
+        Server server = prometheusServer.getServer();
+        server.stop();
+    }
+
+    @Override
+    public void onTrigger(final ReportingContext context) {
+
+        PrometheusServer.context = context;
```

Comment:
What if multiple tasks are configured? I've just tested and it seems they will overwrite each other

Reply:
Made those variables private.